polaris-service/src/main/java/org/apache/polaris/service/exception/IcebergExceptionMapper.java

```diff
@@ -128,9 +128,18 @@ case StorageException e when doesAnyThrowableContainInsensitive(
    * @return whether any throwable in the exception chain case-insensitive-contains the given
    *     message
    */
-  static boolean doesAnyThrowableContainInsensitive(Exception e, String message) {
-    String messageLower = message.toLowerCase(Locale.ENGLISH);
+  static boolean doesAnyThrowableContainAccessDeniedHint(Exception e) {
     return Arrays.stream(ExceptionUtils.getThrowables(e))
-        .anyMatch(t -> t.getMessage().toLowerCase(Locale.ENGLISH).contains(messageLower));
+        .anyMatch(t -> containsAnyAccessDeniedHint(t.getMessage()));
+  }
+
+  public static boolean containsAnyAccessDeniedHint(String message) {
+    String messageLower = message.toLowerCase(Locale.ENGLISH);
+    return Arrays.stream(ACCESS_DENIED_HINTS).anyMatch(messageLower::contains);
```

Comment:
nit: Since we are primarily checking `anyMatch` maybe we should just use a Set

Reply:
Updated to a Set since semantically it is one.